sql/slave.cc

```diff
@@ -4448,13 +4448,11 @@ static int exec_relay_log_event(THD* thd, Relay_log_info* rli,
         duration between that unset and the time that LMT would be updated
         could lead to spikes in SBM.
 
-        The check for queued_count == dequeued_count ensures the worker threads
-        are all idle (i.e. all events have been executed).
+        The check for `rli->worker_threads_caught_up` ensures the worker threads
+        are all practically idle (i.e. all user events have been executed).
       */
       if ((unlikely(rli->last_master_timestamp == 0) ||
-           (rli->sql_thread_caught_up &&
-            (rli->last_inuse_relaylog->queued_count ==
-             rli->last_inuse_relaylog->dequeued_count))) &&
+           (rli->sql_thread_caught_up && rli->worker_threads_caught_up)) &&
           event_can_update_last_master_timestamp(ev))
       {
```

Comment:
This section is curiously similar to the serial branch fifty lines above, but without adding the `ev->exec_time`.
Was it intended?